tutorials/jmsActiveMQ/src/test/java/com/example/jmsActiveMQ/ApplicationTest.java

```diff
@@ -0,0 +1,205 @@
+      LocalMode.Controller lc = asyncRun();
+
+      // check for presence of output file                                                                                                        
+      chkOutput();
+
+      // compare output lines to input                                                                                                            
+      compare();
+      
+      lc.shutdown();
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  /**
+   * Start JMS broker from here
+   *
+   * @throws Exception
+   */
+  private void startJMSService() throws Exception
+  {
+    broker = new BrokerService();
+    String brokerName = "ActiveMQOutputOperator-broker";
+    broker.setBrokerName(brokerName);
+    broker.getPersistenceAdapter().setDirectory(new File("target/activemq-data/" + 
+        broker.getBrokerName() + '/' + 
+        org.apache.activemq.store.kahadb.KahaDBPersistenceAdapter.class.getSimpleName()).getAbsoluteFile());
+    broker.addConnector("tcp://localhost:61617?broker.persistent=false");
+    broker.getSystemUsage().getStoreUsage().setLimit(1024 * 1024 * 1024);  // 1GB
+    broker.getSystemUsage().getTempUsage().setLimit(100 * 1024 * 1024);    // 100MB
+    broker.setDeleteAllMessagesOnStartup(true);
+    broker.start();
+  }
+  
+  private void createAMQClient() throws Exception 
+  {
+    startJMSService();
+    
+    // Create a ConnectionFactory
+    ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory("vm://localhost");
+
+    // Create a Connection
+    connection = connectionFactory.createConnection();
+    connection.start();
+
+    // Create a Session
+    session = connection.createSession(false, Session.CLIENT_ACKNOWLEDGE);
+
+    // Create the destination (Topic or Queue)
+    Destination destination = session.createQueue(QUEUE_NAME);
+
+    // Create a MessageProducer from the Session to the Topic or Queue
+    producer = session.createProducer(destination);
+    producer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+  }
+
+  private void writeMsg(String[] msgs) throws JMSException 
+  {
+    for (String text : msgs) {
+      TextMessage message = session.createTextMessage(text);
+      producer.send(message);
+    }
+  }
+  
+  private void writeToQueue() throws JMSException {
+    writeMsg(lines);
+    
+    // Clean up
+    session.close();
+    connection.close();
+    
+    LOG.debug("Sent messages to topic {}", QUEUE_NAME);
+  }
+  
+  private Configuration getConfig() {
+    Configuration conf = new Configuration(false);
+    conf.set(ActiveMQApplication.QUEUE_NAME_PROPERTY, QUEUE_NAME);
+
+    String pre = "dt.operator.fileOut.prop.";
+    conf.set(   pre + "filePath",        FILE_DIR);
+    conf.set(   pre + "baseName",        FILE_NAME);
+    conf.setInt(pre + "maxLength",       45);
+    conf.setInt(pre + "rotationWindows", 1);
```

Comment:
Please set properties in an XML configuration file (e.g. properties.xml) and source that file here as done in several other examples, for instance: https://github.com/DataTorrent/examples/blob/master/tutorials/jdbcIngest/src/test/java/com/example/mydtapp/JdbcInputAppTest.java

Using properties files is preferred since it shows users how to change configuration values without having to change the code. You can override a couple here and add a comment to say that they can also be set in code, for illustrative purposes.